testing/test_notebooks.py

```diff
@@ -14,8 +15,8 @@
 # changed notebooks. `pytest` does not allow `/` as part of the test name for the -k argument.
 # This also means that the approach is limited to all notebooks being in the `NOTEBOOK_DIR`
 # top-level path.
-NOTEBOOK_DIR = 'examples'
-ALL_NOTEBOOKS = {Path(x).name for x in glob.glob('examples/*.ipynb')}
+NOTEBOOK_DIR = 'doc/source/examples'
+ALL_NOTEBOOKS = {Path(x).name for x in glob.glob(os.path.join(NOTEBOOK_DIR, '*.ipynb'))}
```

Comment:
Hmm is this correct?

Also if we do need this probably easiest to stick with `pathlib`?

Reply:
I think so? If I add a `print(ALL_NOTEBOOKS)` and run `pytest -s testing/test_notebooks.py` I get:

```
{'od_aegmm_kddcup.ipynb', 'ad_ae_cifar10.ipynb', 'cd_distillation_cifar10.ipynb', 'cd_mmd_cifar10.ipynb', 'od_seq2seq_ecg.ipynb', 'alibi_detect_deploy.ipynb', 'cd_clf_cifar10.ipynb', 'od_sr_synth.ipynb', 'cd_ks_cifar10.ipynb', 'od_vae_cifar10.ipynb', 'cd_model_unc_cifar10_wine.ipynb', 'od_prophet_weather.ipynb', 'cd_text_imdb.ipynb', 'cd_spot_the_diff_mnist_wine.ipynb', 'od_llr_mnist.ipynb', 'od_mahalanobis_kddcup.ipynb', 'cd_online_wine.ipynb', 'simple_2d_drift.ipynb', 'cd_online_camelyon.ipynb', 'od_if_kddcup.ipynb', 'od_ae_cifar10.ipynb', 'cd_mol.ipynb', 'cd_chi2ks_adult.ipynb', 'od_vae_adult.ipynb', 'od_vae_kddcup.ipynb', 'od_seq2seq_synth.ipynb', 'od_llr_genome.ipynb', 'cd_text_amazon.ipynb'}

Will change to Pathlib in any case!
```

Reply:
Hmm I need to remind myself but the way it's done currently might be intentional due to the issue with `pytest` not supporting forward slashes on the test selection which we do on the `.yml` file.

Reply:
I think this line clarifies it: basically it should just be the notebook name (no directories) because of the `pytest` limitation: https://github.com/SeldonIO/alibi-detect/blob/4299d90fb9daada035abc4449dc5db0a8bd21f1a/.github/workflows/test_changed_notebooks.yml#L55-L56

Reply:
I'm only appending the directory name in the `glob` call though. `Path(x).name` is still what is stored in the dict i.e. the dict contains only notebook names. I think the behaviour is the same as before? e.g. before we had `glob.glob('examples/*.ipynb')}`?

Reply:
p.s. I think the current error in the notebook tests is actually coming from what you mentioned above. The github workflow is removing `examples` from the path names, but not the new `doc/source`. I'll fix that now and we can check it works!

Reply:
The tests seem to run now, so I think this is resolved.